modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/main/resources/META-INF/resources/select.soy

```diff
@@ -41,31 +43,90 @@
 				{/if}
 			{/if}
 
-			<div class="input-group-container">
-				<select class="form-control" dir="{$dir}" {if $readOnly}disabled{/if} id="{$name}" name="{$name}" {if $multiple}multiple size="{length($options)}"{/if}>
+			<div class="input-group-container form-builder-select-field">
+
+				{let $selectedValue}
 					{if not $readOnly}
-						<option dir="{$dir}" disabled {if length($value) == 0}selected{/if} value="">
-							{$strings.chooseAnOption}
-						</option>
+						{$strings.chooseAnOption}
+					{else}
+						{if $options.length}
+							{$options[0].value}
+						{/if}
 					{/if}
+				{/let}
+
+				{if not $multiple}
+					<a class="select-field-single-chosen" href="javascript:;" dir="{$dir}" id="{$name}" name="{$name}">
+						<span class="option-selected">{$selectedValue}</span>
+					</a>
+				{else}
+					//TODO Multiple select
+				{/if}
 
-					{foreach $option in $options}
-						{let $selected}
-							{foreach $currentValue in $value}
-								{if $currentValue == $option.value}
-									selected
+				{if not $readOnly}
+					<div class="drop-chosen hide">
+						<div class="search-chosen">
+							<div class="select-search-container">
+								{if $selecteSearchIcon}
+									<a class="" href="javascript:;">{$selecteSearchIcon|noAutoescape}</a>
 								{/if}
+							</div>
+							<input placeholder="Search" class="drop-chosen-search" type="text" autocomplete="off">
+						</div>
+						<ul class="results-chosen">
+							{foreach $option in $options}
+								{let $selected}
+									{foreach $currentValue in $value}
+										{if $currentValue == $option.value}
+											selected
+										{/if}
+									{/foreach}
+								{/let}
+								<li class="{$selected ? 'option-selected' : ''}" data-option-index="{index($option)}">
+									{$option.value}
+								</li>
 							{/foreach}
-						{/let}
+						</ul>
+					</div>
+				{/if}
+
+				// OLD VERSION
+				//
+				 <select class="form-control hide" dir="{$dir}" {if $readOnly}disabled{/if} id="{$name}" name="{$name}" {if $multiple}multiple size="{length($options)}"{/if}>
```

Comment:
I think is a good a ideia, this new part is only for the user interaction. But from this way we need to change the option select in the JS, right?